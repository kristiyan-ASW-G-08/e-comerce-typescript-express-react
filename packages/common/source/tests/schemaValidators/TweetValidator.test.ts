import TweetValidator from '@schemaValidators/TweetValidator';

describe('TweetValidator', (): void => {
  const validTweets = [
    {
      type: 'text',
      text:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vel alias, amet corporis modi corrupti.',
    },
    {
      type: 'link',
      text:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vel alias, amet corporis modi corrupti.',
      linkUrl: 'https://fakeLink.fakeLink',
    },
    {
      type: 'retweet',
      text:
        'Lorem ipsum dolor sit amet consectetur adipisicing elit. Similique vel alias, amet corporis modi corrupti.',
      retweetId: 'mockId',
    },
    {
      type: 'reply',
      replyId: 'mockId',
    },
  ];

  const invalidTweets = [
    {
      type: 'invalid',
      text: 10,
    },
    {
      type: 'link',
      text: '',
      linkUrl: "not a url'",
    },
  ];
  it.each(validTweets)('should validate successfully', async tweet => {
    expect.assertions(1);
    await expect(
      TweetValidator.validate(tweet, { abortEarly: false }),
    ).resolves.toBe(tweet);
  });
  it.each(invalidTweets)('should validate unsuccessfully', async tweet => {
    // expect.assertions(1);
    await expect(
      TweetValidator.validate(tweet, { abortEarly: false }),
    ).rejects.toMatchSnapshot();
  });
});
