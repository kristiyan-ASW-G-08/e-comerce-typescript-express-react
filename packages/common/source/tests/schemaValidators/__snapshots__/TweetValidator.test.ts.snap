// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetValidator should validate unsuccessfully 1`] = `[ValidationError: type must be one of the following values: text, link, retweet, reply]`;

exports[`TweetValidator should validate unsuccessfully 2`] = `[ValidationError: linkUrl must be a valid URL]`;
